### PR TITLE
Issues/4094 manage sites header

### PR DESCRIPTION
**Pull request by aerych**
Refs #4094

This PR changes how the managed sites feature is accessed.  The accessory view used previously in the menu is removed and instead a header is presented at the top of the Followed Sites stream. 
If the user is not following any sites, the header is hidden and instead the "no results" view shows a call to action. 

#### Manage Sites Header:
simulator screen shot jul 26 2016 3 10 14 pm

#### No Results:
simulator screen shot jul 26 2016 3 09 33 pm

cc @jancavan, @mattmiklic for a design check
@jleandroperez would you be game for a code review? 

Needs review: @jleandroperez

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesStreamHeader.swift [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+{
+    @IBOutlet private weak var innerContentView: UIView!
+    @IBOutlet private weak var imageView: UIImageView!
+    @IBOutlet private weak var titleLabel: UILabel!
+    @IBOutlet private weak var disclosureIcon: UIImageView!
+    @IBOutlet private weak var contentButton: UIButton!
+    @IBOutlet private weak var contentIPadTopConstraint: NSLayoutConstraint?
+    @IBOutlet private weak var contentBottomConstraint: NSLayoutConstraint!
+
+    public var delegate: ReaderStreamHeaderDelegate?
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesStreamHeader.swift]: Maybe *weak* ?

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesStreamHeader.swift [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+    func applyStyles() {
+        backgroundColor = WPStyleGuide.greyLighten30()
+
+        titleLabel.font = WPStyleGuide.tableviewTextFont()
+        titleLabel.textColor = WPStyleGuide.darkGrey()
+        titleLabel.text = NSLocalizedString("Manage Sites", comment: "Button title. Tapping lets the user manage the sites they follow.")
+
+        disclosureIcon.image = Gridicon.iconOfType(.ChevronRight, withSize: disclosureIcon.frame.size)
+        disclosureIcon.tintColor = UIColor(fromRGBAColorWithRed: 199.0, green: 199.0, blue: 204.0, alpha: 1.0)
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesStreamHeader.swift]: WPStyleGuide?

File: WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesStreamHeader.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+    public func enableLoggedInFeatures(enable: Bool) {
+        // no op
+    }
+
+
+    // MARK: - Actions
+
+
+    @IBAction func didTouchDown(sender: UIButton) {
+        innerContentView.backgroundColor = UIColor(fromRGBAColorWithRed: 217.0, green: 217.0, blue: 217.0, alpha: 1.0)
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderFollowedSitesStreamHeader.swift]: Same as above

File: WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift [hunk 4]
```diff
@@ -1398,11 +1416,16 @@ import WordPressComAnalytics
...
+        guard let topic = readerTopic else {
+            return
+        }
         // Toggle following for the topic
-        if readerTopic!.isKindOfClass(ReaderTagTopic) {
-            toggleFollowingForTag(readerTopic as! ReaderTagTopic)
-        } else if readerTopic!.isKindOfClass(ReaderSiteTopic) {
-            toggleFollowingForSite(readerTopic as! ReaderSiteTopic)
+        if topic.isKindOfClass(ReaderTagTopic) {
+            toggleFollowingForTag(topic as! ReaderTagTopic)
```
Comment (jleandroperez) [inline WordPress/Classes/ViewRelated/Reader/ReaderStreamViewController.swift]: Just wondering if this is the same as:

`if let topic = readerTopic as? ReaderTagTopic { }`

**Comment by jleandroperez** [comment]
simulator screen shot jul 27 2016 11 01 56

Definitely unrelated to this PR, but the `Sites` title in the manage section is off, on iPad devices.

Few nipticky comments added. Other than that, works and looks great!. 

Feel free to :shipit: once the comments are addressed!!

**Comment by jleandroperez** [comment]
Ahhhh sorry, forgot to test the empty state!!. Updated previous comment. Found a small glitch:

1. Open the `Followed Sites`
2. Open the `Manage` section
3. Unfollow all of the sites
4. Hit Back

As a result, the Header will be hidden, but there will still be an empty gap with the header's size.

**Comment by aerych** [comment]
> Definitely unrelated to this PR, but the Sites title in the manage section is off, on iPad devices

@frosty, @kurzee maybe a side effect from the iPad work?  I can patch unless you guys are already aware and working on it.

**Comment by kurzee** [comment]
>@frosty, @kurzee maybe a side effect from the iPad work? I can patch unless you guys are already aware and working on it.

Whoa, I'll take care of that one after this is merged. I think I broke that at the last second on https://github.com/wordpress-mobile/WordPress-iOS/pull/5764

**Comment by aerych** [comment]
> but there will still be an empty gap with the header's size.

Hola @jleandroperez!  Thanks for the review sir.  Code comments addressed.  I'm trying to reproduce `but there will still be an empty gap with the header's size.` but things look correct to me, both on the iPhone and iPad so I must be missing something.  Do you have a screen shot you could share?

**Comment by jleandroperez** [comment]
@aerych fix confirmed, looking great!

:shipit: on my side sir!

**Comment by jancavan** [comment]
@aerych Minor comment: I think the cog icon is rather large. Can we make it about the same size as the globe icon?

screenshot-2016-08-02-16 25 29

This one has been merged so perhaps for a separate PR.

**Comment by aerych** [comment]
Will do @jancavan !

**Pull request merged**